src/index.ts

```diff
@@ -111,17 +115,25 @@ export default class implements ConnectPlugin {
         return this.stories.length > 0;
     }
 
-    private createLink(kind: string, name?: string): Link {
-        const urlEncodedKind = encodeURIComponent(kind);
+    private createLink(params: { storyId?: string; kind: string; name?: string }): Link {
+        const { storyId, kind, name } = params;
 
-        let preparedUrl;
-        if (name) {
-            const urlEncodedName = encodeURIComponent(name);
-            preparedUrl = urljoin(this.targetUrl, `?selectedKind=${urlEncodedKind}&selectedStory=${urlEncodedName}`);
+        let url: string;
+        if (storyId) {
+            url = join(this.targetUrl, `/?path=/story/${storyId}`);
         } else {
-            preparedUrl = urljoin(this.targetUrl, `?selectedKind=${urlEncodedKind}`);
+            const encodedKind = encodeURIComponent(kind);
+
+            if (name) {
+                url = join(
+                    this.targetUrl,
+                    `/?selectedKind=${encodedKind}&selectedStory=${encodeURIComponent(name)}`
+                );
+            } else {
+                url = join(this.targetUrl, `/?selectedKind=${encodedKind}`);
```

Comment:
we can use `query` parameter like: `urlJoin(this.targetUrl, { query: { selectedKind: encodedKind } });`

Reply:
I don't think this is necessary. 

Firstly, it just makes me to repeat put same option inputs on every call (proper-url-join has troublesome API definition, the problem is not type definition but the implementation is not suitable for rest params input definition).

Secondly, it is more visible to maintenance. No need to delegate SB URL specifics into a 3rd party lib. I just use it to ensure the final concatenated string is a good lookin valid URL without redundant characters(e.g. excessive slashes).

Reply:
👌